library/src/com/google/maps/android/kml/KmlRenderer.java

```diff
@@ -0,0 +1,870 @@
+        }
+        if (inlineStyle.isLineRandomColorMode()) {
+            polylineOptions.color(computeRandomColor(inlinePolylineOptions.getColor()));
+        }
+    }
+
+    /**
+     * Adds a KML Polygon to the map as a Polygon by combining the styling and coordinates
+     *
+     * @param polygon contains coordinates for the Polygon
+     * @param style   contains relevant styling properties for the Polygon
+     * @return Polygon object
+     */
+    private Polygon addPolygonToMap(KmlPolygon polygon, KmlStyle style, KmlStyle inlineStyle) {
+        PolygonOptions polygonOptions = style.getPolygonOptions();
+        polygonOptions.addAll(polygon.getOuterBoundaryCoordinates());
+        for (ArrayList<LatLng> innerBoundary : polygon.getInnerBoundaryCoordinates()) {
+            polygonOptions.addHole(innerBoundary);
+        }
+        if (inlineStyle != null) {
+            setInlinePolygonStyle(polygonOptions, inlineStyle);
+        } else if (style.isPolyRandomColorMode()) {
+            polygonOptions.fillColor(computeRandomColor(polygonOptions.getFillColor()));
+        }
+        return mMap.addPolygon(polygonOptions);
+    }
+
+    /**
+     * Sets the inline polygon style by copying over the styles that have been set
+     *
+     * @param polygonOptions polygon options object to add inline styles to
+     * @param inlineStyle    inline styles to apply
+     */
+    private void setInlinePolygonStyle(PolygonOptions polygonOptions, KmlStyle inlineStyle) {
+        PolygonOptions inlinePolygonOptions = inlineStyle.getPolygonOptions();
+        if (inlineStyle.hasFill() && inlineStyle.isStyleSet("fillColor")) {
+            polygonOptions.fillColor(inlinePolygonOptions.getFillColor());
+        }
+        if (inlineStyle.hasOutline()) {
+            if (inlineStyle.isStyleSet("outlineColor")) {
+                polygonOptions.strokeColor(inlinePolygonOptions.getStrokeColor());
+            }
+            if (inlineStyle.isStyleSet("width")) {
+                polygonOptions.strokeWidth(inlinePolygonOptions.getStrokeWidth());
+            }
+        }
+        if (inlineStyle.isPolyRandomColorMode()) {
+            polygonOptions.fillColor(computeRandomColor(inlinePolygonOptions.getFillColor()));
+        }
+    }
+
+    /**
+     * Adds all the geometries within a KML MultiGeometry to the map. Supports recursive
+     * MultiGeometry. Combines styling of the placemark with the coordinates of each geometry.
+     *
+     * @param multiGeometry contains array of geometries for the MultiGeometry
+     * @param style         contains relevant styling properties for the MultiGeometry
+     * @return array of Marker, Polyline and Polygon objects
+     */
+    private ArrayList<Object> addMultiGeometryToMap(KmlPlacemark placemark,
+            KmlMultiGeometry multiGeometry, KmlStyle style, KmlStyle inlineStyle,
+            Boolean isVisible) {
+        ArrayList<Object> geometries = new ArrayList<Object>();
```

Comment:
Unclear of what this represents, should rename to mapObjects or something similar